jscomp/test/record_regression.res

```diff
@@ -100,3 +100,42 @@ let setAA = (ao: option<int>) => {aa: @ns.optional ao, bb: None}
 //     @ns.optional bb: int,
 //   }
 // }
+
+type foo = Foo({name: string, age?: int})
+
+let foo1 = Foo({name: "foo"}) 
+let foo2 = Foo({name: "foo", age: 3})
+
+// should be type error
+// let foo3 = Foo({name: "foo", age: 3, nickname: "hasNoNickname"})
+// let foo4 = Foo({name: "foo", age: "3"})
+
+type inlinedRecord = V0({x0: string, x1?: string, x2?: int, x3: int}) | V1({y0: string, y1: int})
+
+let ir0 = V0({x0: "v0", x3: 3})
+let ir1 = V0({x0: "v0", x1: "v1", x3: 3})
+let ir2 = V0({x0: "v0", x1: "v1", x2: 2, x3: 3})
+let ir3 = V1({y0: "v0", y1: 1})
+
+let pm0 = switch ir0 {
+  | V0({x0, x3}) => (x0, x3)
+  | V1({y0, y1}) => (y0, y1)
+}
+let pm1 = switch ir1 {
+  | V0({x0, x1, x3}) => (x0, x1, x3)
```

Comment:
This is not quite right: `x1` here should not have option type.
Check this basic test for records: https://rescript-lang.org/try?code=C4TwDgpgBATlC8UDeAPAXFAlgO2AGihAH4MBnYGHAcwF8AoOgGwmCggEYFl0oBme5qwgAmLqgy8CIDACIAhgCMAxjIEs2vMT0mEMRAHIB7bBHpN1YTojjwAfHShRSAd0zAlAC1jIHUAD7IIDQItoRQANThUDIyvgFI0lAGxqYh0bGO9EA

Inline records should behave the same.

Reply:
Check also the docs: https://rescript-lang.org/docs/manual/latest/record#pattern-matching-on-optional-fields

Reply:
There's likely some processing missing in the type checker. Which is done for records, but not for inline ones.

Reply:
Ah! Got your point! I should need to check `matching.ml` Let me fix it.

Reply:
Not `matching.ml`, it is issue related to `pat_record`.

Reply:
Yes look for `process_optional_label`.

Reply:
It reminds me of how our compiler is so smart! https://github.com/rescript-lang/rescript-compiler/pull/5827/commits/cd876adb772008ac69ad7e60a2a842e50075622e